psycopg/psycopg/_pipeline.py

```diff
@@ -63,36 +64,12 @@ def status(self) -> pq.PipelineStatus:
     def is_supported(cls) -> bool:
         """Return `!True` if the psycopg libpq wrapper supports pipeline mode."""
         if BasePipeline._is_supported is None:
-            BasePipeline._is_supported = not cls._not_supported_reason()
+            BasePipeline._is_supported = capabilities.has_pipeline()
         return BasePipeline._is_supported
 
-    @classmethod
-    def _not_supported_reason(cls) -> str:
-        """Return the reason why the pipeline mode is not supported.
-
-        Return an empty string if pipeline mode is supported.
-        """
-        # Support only depends on the libpq functions available in the pq
-        # wrapper, not on the database version.
-        if pq.version() < 140000:
-            return (
-                f"libpq too old {pq.version()};"
-                " v14 or greater required for pipeline mode"
-            )
-
-        if pq.__build_version__ < 140000:
-            return (
-                f"libpq too old: module built for {pq.__build_version__};"
-                " v14 or greater required for pipeline mode"
-            )
-
-        return ""
-
     def _enter_gen(self) -> PQGen[None]:
-        if not self.is_supported():
-            raise e.NotSupportedError(
-                f"pipeline mode not supported: {self._not_supported_reason()}"
-            )
+        if not self._is_supported:
+            capabilities.has_pipeline(check=True)
```

Comment:
IIUC, we'd now always run `capabilities.has_pipeline(check=True)` when entering the pipeline, unless `is_supported()` has been called before to set the `_is_supported` attribute, no?

Also, it might interesting to only call `capabilities.has_pipeline()` once; something like 663e64265cb9aa422aa401e03877ab063f6ef08f.

Reply:
Yes, that's xorrect. The if is a micro optimization to avoid to call `has_pipeline()` every time in the most common path (when we already established that pipelines are supported).

Maybe it's worth nothing, or maybe in general I should have added caching in the Capabilities object, if my gripe is to avoid to call `pq.version()` over and over. 

TBH I've been a bit lazy here and I didn't add caching because I thought it'd make tests bit more cumbersome... But maybe I should do that.